lib/vsc/filesystem/quota/tools.py

```diff
@@ -115,11 +121,46 @@ def push(self, storage_name, payload):
             self.count[storage_name] = 0
             self.payload[storage_name] = []
 
+    def push_quota(self, owner, fileset, quota, shared=False):
+        """
+        Push quota to accountpage: it belongs to owner (can either be user_id or vo_id),
+        in the given fileset and quota.
+        :param owner: the name of the user or VO to which the quota belongs
+        :param fileset: fileset name
+        :param quota: actual quota data
+        :param shared: is this a shared user/VO quota or not?
+        """
+        params = {
+            "fileset": fileset,
+            "used": quota.used,
+            "soft": quota.soft,
+            "hard": quota.hard,
+            "doubt": quota.doubt,
+            "expired": quota.expired[0],
+            "remaining": quota.expired[1] or 0,  # seconds
+            "files_used": quota.files_used,
+            "files_soft": quota.files_soft,
+            "files_hard": quota.files_hard,
+            "files_doubt": quota.files_doubt,
+            "files_expired": quota.files_expired[0],
+            "files_remaining": quota.files_expired[1] or 0,  # seconds
+        }
+
+        if self.kind == QUOTA_USER_KIND:
+            params['user'] = owner
+        elif self.kind == QUOTA_VO_KIND:
+            params['vo'] = owner
+
```

Comment:
else: raise Exception?

Reply:
That's already done on the actual push to the accountpage.